KC_Liberation_Master_Framework/karmakut/trash_cleanup/server/init.sqf

```diff
@@ -0,0 +1,48 @@
+karma_trashCleanup_server_trashToProcessPerIteration = 25;
+karma_trashCleanup_server_trashContainersToProcess = []; // Element structure: [_trashContainer, _creationTimestamp]
+karma_trashCleanup_server_queuedTrashContainers = [];
+karma_trashCleanup_server_queuedTrashContainersLock = false; // Used for preventing writes from interfering between queueing and processing.
+
+karma_trashCleanup_server_registerTrashContainer = {
+    params [
+        ["_trashContainer", objNull]
+    ];
+    if (isNull _trashContainer) exitWith {};
+    waitUntil { !karma_trashCleanup_server_queuedTrashContainersLock };
+    karma_trashCleanup_server_queuedTrashContainersLock = true;
+    karma_trashCleanup_server_queuedTrashContainers pushBack [_trashContainer, time];
+    karma_trashCleanup_server_queuedTrashContainersLock = false;
+};
+
+[] spawn {
+    while { true } do {
+        private _time = time;
+        private _trashIteration = 0;
+        {
+            private _trashContainer = _x select 0;
+            private _creationTimestamp = _x select 1;
+            private _isTrashOlderThanAllowedAge = _time - _creationTimestamp >= karma_trashCleanup_shared_trashRemovalThresholdInSeconds;
```

Comment:
Can move this down between L26 and L27. This variable is useless if the _trashContainer is null.